### PR TITLE
Additional check for checkbox in emptySelectedCells for cut action

### PR DESCRIPTION
**Pull request by pnowak**
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #4106

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/editorManager.js [hunk 1]
```diff
@@ -402,7 +402,7 @@ class EditorManager {
 
       case KEY_CODES.BACKSPACE:
       case KEY_CODES.DELETE:
-        this.instance.emptySelectedCells();
+        this.instance.emptySelectedCells('delete');
```
Comment (swistach) [inline src/editorManager.js]: I think we can leave it at it is, `delete` we process as just `edit` action.
```suggestion
        this.instance.emptySelectedCells();
```

File: src/plugins/copyPaste/copyPaste.js [hunk 2]
```diff
@@ -458,7 +458,7 @@ class CopyPaste extends BasePlugin {
         this.hot.rootWindow.clipboardData.setData('Text', textPlain);
       }
 
-      this.hot.emptySelectedCells();
+      this.hot.emptySelectedCells('cut');
```
Comment (swistach) [inline src/plugins/copyPaste/copyPaste.js]:
```suggestion
      this.hot.emptySelectedCells('CopyPaste.cut');
```

File: src/renderers/checkboxRenderer.js [hunk 3]
```diff
@@ -51,14 +51,17 @@ function checkboxRenderer(instance, TD, row, col, prop, value, cellProperties, .
 
   } else {
     input.style.display = 'none';
-    addClass(input, BAD_VALUE_CLASS);
-    badValue = true;
+
```
Comment (swistach) [inline src/renderers/checkboxRenderer.js]:
```suggestion
```

File: src/renderers/checkboxRenderer.js [hunk 4]
```diff
@@ -51,14 +51,17 @@ function checkboxRenderer(instance, TD, row, col, prop, value, cellProperties, .
...
+    if (cellProperties.onCut !== true) {
```
Comment (swistach) [inline src/renderers/checkboxRenderer.js]:
```suggestion
```

File: src/renderers/checkboxRenderer.js [hunk 5]
```diff
@@ -51,14 +51,17 @@ function checkboxRenderer(instance, TD, row, col, prop, value, cellProperties, .
...
+      addClass(input, BAD_VALUE_CLASS);
```
Comment (swistach) [inline src/renderers/checkboxRenderer.js]:
```suggestion
```

File: src/renderers/checkboxRenderer.js [hunk 6]
```diff
@@ -51,14 +51,17 @@ function checkboxRenderer(instance, TD, row, col, prop, value, cellProperties, .
...
+      badValue = true;
```
Comment (swistach) [inline src/renderers/checkboxRenderer.js]:
```suggestion
```

File: src/renderers/checkboxRenderer.js [hunk 7]
```diff
@@ -51,14 +51,17 @@ function checkboxRenderer(instance, TD, row, col, prop, value, cellProperties, .
...
+    }
```
Comment (swistach) [inline src/renderers/checkboxRenderer.js]:
```suggestion
```

File: src/renderers/checkboxRenderer.js [hunk 8]
```diff
@@ -51,14 +51,17 @@ function checkboxRenderer(instance, TD, row, col, prop, value, cellProperties, .
...
   }
 
   input.setAttribute('data-row', row);
   input.setAttribute('data-col', col);
 
-  if (!badValue && labelOptions) {
+  if (!badValue && !cellProperties.onCut && labelOptions) {
```
Comment (swistach) [inline src/renderers/checkboxRenderer.js]:
```suggestion
  if (!badValue && labelOptions) {
```

File: test/e2e/renderers/checkboxRenderer.spec.js [hunk 9]
```diff
@@ -12,6 +12,30 @@ describe('CheckboxRenderer', () => {
     }
   });
 
+  class DataTransferObject {
```
Comment (swistach) [inline test/e2e/renderers/checkboxRenderer.spec.js]: Could you move `DataTransferObject` and `getClipboardEvent` to common helper? CopyPaste's tests could use it too.

File: test/helpers/common.js [hunk 10]
```diff
@@ -776,3 +776,27 @@ export function triggerTouchEvent(type, target, pageX, pageY) {
 export function createSpreadsheetData(...args) {
   return Handsontable.helper.createSpreadsheetData(...args);
 }
+
+export class DataTransferObject {
```
Comment (swistach) [inline test/helpers/common.js]: I think that we don't need to export this class.
```suggestion
class DataTransferObject {
```

File: test/e2e/renderers/checkboxRenderer.spec.js [hunk 11]
```diff
@@ -818,6 +818,84 @@ describe('CheckboxRenderer', () => {
...
+          checkedTemplate: 'yes',
+          uncheckedTemplate: 'no',
+        },
+      ],
+    });
+    const cutEvent = getClipboardEvent('cut');
+    const plugin = hot.getPlugin('CopyPaste');
+    const td = hot.getCell(0, 2);
+
+    selectCell(0, 2);
```
Comment (swistach) [inline test/e2e/renderers/checkboxRenderer.spec.js]: Could you select all cells and verify how it works with both cases of using the checkbox type?

Comment (pnowak) [reply test/e2e/renderers/checkboxRenderer.spec.js]: Yes, I wil try.

**Pull request merged**